proto/info/info.proto

```diff
@@ -1316,3 +1319,34 @@ message ZInfoLocation {
   string logicallabel = 9;
 }
 
+message KubeClusterUpgradeStatus {
+  // current_node will have an empty value when no node is in an upgrade
+  string current_node = 1;
+
+  // component currently under upgrade, COMP_UNKNOWN when no upgrades in progress
+  KubeComp component = 2;
+
+  // status of the current component under upgrade, KUBE_COMP_UPGRADE_STATUS_UNKNOWN when
+  //   no upgrades in progress
+  KubeCompUpgradeStatus status = 3;
+
+  // Error info in case of failure
+  ErrorInfo error = 4;
+}
+
+message KubeClusterInfo {
+  // Info message on a list of cluster nodes
+  repeated KubeNodeInfo nodes = 1;
+
+  // Info message on a list of namespaces's pods summary
+  repeated KubePodNameSpaceInfo pod_name_spaces = 2;
+
+  // Info message on a list of EVE applications
+  repeated KubeEVEAppPodInfo eve_apps = 3;
+
+  // Info message on cluster storage
+  KubeStorageInfo storage = 4;
+
+  // The current status of the cluster upgrade
+  KubeClusterUpgradeStatus upgrade = 5;
```

Comment:
What will this 'upgrade' report as part of the KubeClusterInfo ?

Reply:
It would report unspecified or the last component updated but this is a good point, it might be appropriate to remove it from KubeClusterInfo.

Reply:
latest change removes upgrade from KubeClusterInfo